cmd/clusterctl/client/alpha/rollout_restarter.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpha
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var validResourceTypes = []string{"machinedeployment"}
+
+// ObjectRestarter will issue a restart on the specified cluster-api resource.
+func (r *rollout) ObjectRestarter(proxy cluster.Proxy, tuple util.ResourceTuple, namespace string) error {
+	switch tuple.Resource {
+	case "machinedeployment":
+		mdObj, err := getMachineDeployment(proxy, tuple.Name, namespace)
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch %v/%v", tuple.Resource, tuple.Name)
+		}
+		if mdObj == nil {
+			return errors.Errorf("failed to fetch %v/%v", tuple.Resource, tuple.Name)
+		}
```

Comment:
Apologies for not noticing this earlier. But this is a minor refactor.

```suggestion
		if err != nil || mdObj == nil {
			return errors.Wrapf(err, "failed to fetch %v/%v", tuple.Resource, tuple.Name)
		}
```